src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,164 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "valid_login";
+    private static final String VALID_LOGIN_EDGE_CASE = "login6";
+    private static final String VALID_PASSWORD = "valid_password";
+    private static final String VALID_PASSWORD_EDGE_CASE = "123456";
+    private static final int VALID_AGE = 25;
+    private static final int VALID_AGE_EDGE_CASE = 18;
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+    }
+
+    @Test
+    void register_allValidData_Ok() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        User expected = storageDao.add(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_allValidDataEdgeCase_Ok() {
+        user.setLogin(VALID_LOGIN_EDGE_CASE);
+        user.setPassword(VALID_PASSWORD_EDGE_CASE);
+        user.setAge(VALID_AGE_EDGE_CASE);
+        User expected = storageDao.add(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        User user2 = new User();
+        user2.setLogin(VALID_LOGIN);
+        user2.setPassword(VALID_PASSWORD_EDGE_CASE);
+        user2.setAge(VALID_AGE);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));

```

Comment:
I would recommend adding an informative message to each test, for example "the method should throw a RegistrationException if the user with this login is already registered"

Reply:
I don't think it's necessary

Reply:
It would be great to test which message you received.